sklearn/covariance/_empirical_covariance.py

```diff
@@ -103,10 +103,8 @@ def empirical_covariance(X, *, assume_centered=False):
             "Only one sample available. You may want to reshape your data array"
         )
```

Comment:
To avoid `nan`, we can directly return an array of zero for a single sample.

```suggestion
        )
        return np.zeros((X.shape[1], X.shape[1]), dtype=X.dtype)
```